pkg/controller/controller.go

```diff
@@ -0,0 +1,63 @@
+package controller
+
+import (
+	"sync"
+
+	"github.com/submariner-io/admiral/pkg/federate"
+	"k8s.io/klog"
+
+	"github.com/submariner-io/coastguard/pkg/healthz"
+	"github.com/submariner-io/coastguard/pkg/remotecluster"
+)
+
+// arbitrary event channel size
+const eventChannelSize = 1000
+
+type CoastguardController struct {
+	federate.ClusterEventHandler
```

Comment:
woops, is it a copy? then I don't understand how it works, but it's working. We're missing something.

Reply:
It was working in my case because all the structure attributes are pointers, and previously initializer, but there's no future guarantee to stay like that

Reply:
So an interface is already a pointer that references an internal object that holds the underlying object and type.  This is passed by value in functions. So from inside `WatchClusters`, the address of the `handler` variable actually represents Go's internal representation of the interface. So the actual interface object that you pass is not copied.

Reply:
> I like how you integrated the interface into your struct. 

But this isn't really what you want in this case. The `CoastguardController` already implements the `federate.ClusterEventHandler` interface and I assume you intend to pass the `CoastguardController` instance to `WatchClusters`.  

Embedding `federate.ClusterEventHandler` declares a field of type `federate.ClusterEventHandler` that you would need to initialize in order to use, eg

```
controller := &CoastguardController{
    ClusterEventHandler: someClusterEventHandlerInstance,
    remoteClusters:  make(map[string]*remotecluster.RemoteCluster),
    ...
}
```
 Given the above, then you would do `federator.WatchClusters(controller.ClusterEventHandler)`. So unless you plan to use the embedded `federate.ClusterEventHandler` field then it doesn't add any value (it's effectively an unused field).

Reply:
Internally it's a pointer, I tested it inside/out. No need to change it.

Reply:
> Internally it's a pointer, I tested it inside/out. No need to change it.

Embedding `federate.ClusterEventHandler` doesn't hurt anything but, as I mentioned, it is an unused field and doesn't add an value.

Reply:
> The `CoastguardController` already implements the `federate.ClusterEventHandler` interface and I assume you intend to pass the `CoastguardController` instance to `WatchClusters`.

Yeah, and actually you should include that as a test for the compiler:

`var _ federate.ClusterEventHandler = &CoastguardController{}`